ocaml/xapi/xapi_pusb.ml

```diff
@@ -48,10 +48,17 @@ let scan_start ~__context usbs =
   List.filter (fun (rf, rc) -> USBSet.mem (extract_known_usb_info rc) (USBSet.diff known_usb local_usb)) (Db.PUSB.get_all_records ~__context)
   |> List.iter (fun (self, _) ->
                   let usb_group = Db.PUSB.get_USB_group ~__context ~self in
-                  Db.PUSB.destroy ~__context ~self;
                   let vusbs = Db.USB_group.get_VUSBs ~__context ~self:usb_group in
-                  List.iter (fun vusb -> Db.VUSB.destroy ~__context ~self:vusb) vusbs;
-                  Db.USB_group.destroy ~__context ~self:usb_group)
+                  List.iter (fun vusb ->
+                    let currently_attached = Db.VUSB.get_currently_attached ~__context ~self:vusb in
+                    if currently_attached then begin
```

Comment:
I'd like to catch errors here and continue - so if we fail to unplug the VUSB, we should not destroy the PUSB and USB_group, but we should then continue to process the rest of the scan result.

Reply:
Ok.  make sense

Reply:
updated. please check if the error is ok. Thanks